signer/src/transaction_coordinator.rs

```diff
@@ -2027,4 +2060,45 @@ mod tests {
         // Assert the result
         assert_eq!(result, should_allow);
     }
+
+    fn test_pubkey(seed: u64) -> PublicKey {
+        let mut rng = rand::rngs::StdRng::seed_from_u64(seed);
+        PublicKey::from_private_key(&PrivateKey::new(&mut rng))
+    }
+
+    #[test_case(model::EncryptedDkgShares{dkg_shares_status: model::DkgSharesStatus::Unverified, ..Faker.fake()}, None, true, true; "unverified, no key")]
+    #[test_case(model::EncryptedDkgShares{dkg_shares_status: model::DkgSharesStatus::Verified, ..Faker.fake()}, None, true, true; "verified, no key")]
+    #[test_case(model::EncryptedDkgShares{dkg_shares_status: model::DkgSharesStatus::Unverified, aggregate_key: test_pubkey(1), ..Faker.fake()}, Some(test_pubkey(1)), true, false; "unverified, key up to date")]
+    #[test_case(model::EncryptedDkgShares{dkg_shares_status: model::DkgSharesStatus::Verified, aggregate_key: test_pubkey(1), ..Faker.fake()}, Some(test_pubkey(1)), false, false; "verified, key up to date")]
+    #[test_case(model::EncryptedDkgShares{dkg_shares_status: model::DkgSharesStatus::Unverified, aggregate_key: test_pubkey(2), ..Faker.fake()}, Some(test_pubkey(1)), true, true; "unverified, new key")]
+    #[test_case(model::EncryptedDkgShares{dkg_shares_status: model::DkgSharesStatus::Verified, aggregate_key: test_pubkey(2), ..Faker.fake()}, Some(test_pubkey(1)), true, true; "verified, new key")]
```

Comment:
These are a little hard to read, but it's okay.

Reply:
I was supposed to at least fmt those but I forgot, refactored a bit to improve readability